portfolio/src/main/java/com/google/sps/servlets/NewCommentsServlet.java

```diff
@@ -0,0 +1,42 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.util.Date;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/new-comments")
+public class NewCommentsServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String content = getParameter(request, "text-comment", "this message was empty...");
+    String userName = getParameter(request, "text-name", "anonymous");
+    boolean isAnonymous = Boolean.parseBoolean(getParameter(request, "anonymous", "false"));
+    userName = isAnonymous ? "anonymous" : userName;
+
+    Entity commentEntity = new Entity("Comment");
```

Comment:
A suggestion, create a factory method for a comment object with the minimal number of inputs (content, userName isAnonymous). Initialize all the default fields in the factory and then create a toEntity method for the comment that will produce the entity object for you.

You can also create CommentEntityConverter.java that will do the conversion if you want to keep the Comment class slim.

Reply:
Good idea! I'll keep that in mind next time. Is this what you meant?
```
public static Entity toEntity(String userName, String content) {
    Entity commentEntity = new Entity("Comment");

    commentEntity.setProperty("userName", userName);
    commentEntity.setProperty("content", content);
    commentEntity.setProperty("timestamp", new Date());
    commentEntity.setProperty("like", 0);

    return commentEntity;
  }
```